api/v1alpha1/edgeplaybook_types.go

```diff
@@ -0,0 +1,116 @@
+
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+
+	LogCollection map[string]*LogCollectionConfig `json:"logCollection,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// Ansible binary file
+	BinaryPath string `json:"binaryPath,omitempty"`
+	// the ansible's playbooks list to be used
+	Playbooks []Playbook `json:"playbooks,omitempty"`
+	// the ansible's playbook options
+	Options *AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options
+	PrivilegeEscalationOptions *PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+	// enum that defines which is the stdout callback method (default = 'default'). Supported stdout methods are: debug,default,dense,json,minimal,null,oneline,stderr,timer,yaml,runner-events
+	StdoutCallback string `json:"stdoutCallback,omitempty"`
+}
+
+type AnsibleOptions struct {
+	// don't make any changes; instead, try to predict some of the changes that may occur
+	Check bool `json:"check,omitempty"`
+}
+
+type PrivilegeEscalationOptions struct {
+	Become bool `json:"become,omitempty"`
+	// Enum ksu, pbrun, enable, sesu, pmrun,runas, sudo, su, doas, pfexec, Machinectl, dzdo
+	BecomeMethod string `json:"becomeMethod,omitempty"`
+	BecomeUser   string `json:"becomeUser,omitempty"`
+}
+
+type Playbook struct {
+	// Playbook content
+	Content string `json:"content"`
+	// The connection timeout on ansible-playbook
+	Timeout uint64 `json:"timeout,omitempty"`
+	// The required privelege level necessary to execute the playbook TODO: Enum like linux capabilities ?
+	RequiredPrivilegeLevel string `json:"requiredPrivilegeLevel,omitempty"`
+}
+
+// EdgePlaybookStatus defines the observed state of EdgePlaybook
+type EdgePlaybookStatus struct {
+	Phase                     string               `json:"phase,omitempty"`
+	LastSeenTime              metav1.Time          `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string               `json:"lastSyncedResourceVersion,omitempty"`
+	Hardware                  *Hardware            `json:"hardware,omitempty"`
+	Deployments               []PlaybookDeployment `json:"playbookDeployments,omitempty"`
+	DataOBC                   *string              `json:"dataObc,omitempty"`
+	UpgradeInformation        *UpgradeInformation  `json:"upgradeInformation,omitempty"`
+}
+
+type PlaybookDeployment struct {
+	Name               string              `json:"name"`
+	Phase              EdgeDeploymentPhase `json:"phase,omitempty"`
+	LastTransitionTime metav1.Time         `json:"lastTransitionTime,omitempty"`
+	LastDataUpload     metav1.Time         `json:"lastDataUpload,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// EdgePlaybook is the Schema for the EdgePlaybooks API
+type EdgePlaybook struct {
```

Comment:
I just throwing an idea (don't want to cause too much noise to the PR), this name can be generalized to something such as `EdgeConfig` and now we support only configuration by ansible.

But if there will be another need for an additional method, this will remain fairly easy to extend.
It requires adding one level to the spec saying ansible, e.g.:
```
kind: EdgeConfig
...
spec:
  ansible:
    ... // EdgePlaybookSpec
  other_method: 
     ...
status:
   ...
```
we don't have any need for an alternative method for now, but we'd might want to have that option available (@rgolangh is exploring this)

Reply:
I like that idea.